File: Codeforces/1333B.py
t = int(input())

for i in range(t):
    n = int(input())
    a = list([int(x) for x in input().split()])
    prev = [set()]
    for i in range(1,n):
        s = set()
        s.add(a[i-1])
        s = s.union(prev[i-1])
        prev.append(s)
    b = list([int(x) for x in input().split()])

    if(a[0]!=b[0]):
        print("NO")
        break
    
    flag = True 

    for j in range(1,n):
        if a[j] != b[j]:
            diff = b[j]-a[j]
            if diff>0:
                if(1 in prev[j]):
                    continue
                else:
                    print("NO")
                    flag = False
                    break
            if diff<0:
                if(-1 in prev[j]):
                    continue
                else:
                    print("NO")
                    flag = False
                    break
    if(flag):
        print("YES")

File: Codeforces/1333B.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin >> t;
    for (int k = 0; k < t; k++)
    {
        int n;
        cin >> n;
        vector<int> a;
        for (int i = 0; i < n; i++)
        {
            int val;
            cin >> val;
            a.push_back(val);
        }
        vector<set<int>> prev;
        set<int> init;
        prev.push_back(init);

        for (int i = 1; i < a.size(); i++)
        {
            set<int> new_set;
            new_set.insert(a[i - 1]);
            new_set.insert(prev[i - 1].begin(),prev[i-1].end());
            prev.push_back(new_set);
        }
        vector<int> b;
        for (int i = 0; i < n; i++)
        {
            int val;
            cin >> val;
            b.push_back(val);
        }

        if (a[0] != b[0])
        {
            cout << "NO" << endl;
            continue;
        }

        bool flag = true;

        for (int i = 1; i < n; i++)
        {
            if (a[i] != b[i])
            {
                int diff = b[i] - a[i];
                if (diff > 0)
                {
                    if (prev[i].find(1) != prev[i].end())
                    {
                        continue;
                    }
                    else
                    {
                        flag = false;
                        cout << "NO" << endl;
                        break;
                    }
                }
                else if (diff < 0)
                {
                    if (prev[i].find(-1) != prev[i].end())
                    {
                        continue;
                    }
                    else
                    {
                        flag = false;
                        cout << "NO" << endl;
                        break;
                    }
                }
            }
        }

        if (flag == true)
        {
            cout << "YES" << endl;
        }
    }
    return 0;
}